server/mqtt.go

```diff
@@ -1701,9 +1705,17 @@ func (jsa *mqttJSA) createDurableConsumer(cfg *CreateConsumerRequest) (*JSApiCon
 	return ccr, ccr.ToError()
 }
 
-func (jsa *mqttJSA) deleteConsumer(streamName, consName string) (*JSApiConsumerDeleteResponse, error) {
+// if noWait is specified, does not wait for the JS response, returns nil
+func (jsa *mqttJSA) deleteConsumer(streamName, consName string, noWait bool) (*JSApiConsumerDeleteResponse, error) {
 	subj := fmt.Sprintf(JSApiConsumerDeleteT, streamName, consName)
-	cdri, err := jsa.newRequest(mqttJSAConsumerDel, subj, 0, nil)
+
+	timeout := mqttJSAPITimeout
+	if noWait {
+		timeout = 0
+	}
+
+	// timeout == 0 signals that we don't want to wait for the response.
+	cdri, err := jsa.newRequestEx(mqttJSAConsumerDel, subj, _EMPTY_, -1, nil, timeout)
```

Comment:
This would return "invalid number of responses", which is not quite good.

Reply:
In other words, the original approach may have been ok, you would have simply needed to remove the test that was no longer relevant...

Reply:
Right. Apologies for the confusion.